tests/unit/swagger/definitionsGeneration/typeResolver.spec.ts

```diff
@@ -0,0 +1,512 @@
+import { expect } from 'chai';
```

Comment:
I'd prefer if you could integrate the tests in this file with the existing tests

Reply:
Spec Generators Tests most likely

Reply:
This isn't really about spec-generation though. These tests are about type resolution. The spec generation tests are relevant after a type has already been resolved. None of them use the type resolver.

Currently the testing for type resolution is spread across the controller/method/parameter generator specs, but those are higher order integration tests than what I'm testing here. 
If we were going to merge them with existing tests, it would be somewhere in those. 

In my opinion the lack of easy testability for type resolutions is one of the reasons that there are quite a few type resolution bugs outstanding for the repo. This test may make it easier for folks to identify and fix those in a much more narrow focus area.